provider/azure/azure_discover.go

```diff
@@ -144,19 +149,21 @@ func fetchAddrsWithTags(tagName string, tagValue string, vmnet network.Interface
 
 func fetchAddrsWithVmScaleSet(resourceGroup string, vmScaleSet string, vmnet network.InterfacesClient, l *log.Logger) ([]string, error) {
 	// Get all network interfaces for a specific virtual machine scale set
-	netres, err := vmnet.ListVirtualMachineScaleSetNetworkInterfaces(resourceGroup, vmScaleSet)
+	ctx, cancel := context.WithCancel(context.Background())
```

Comment:
You can just use `context.Background()` with these as there's no concurrency and therefore nothing can ever call `cancel()` in a useful way.